pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -869,6 +877,131 @@ func afwGetReleventStatefulsetByKind(cr splcommon.MetaObject, client splcommon.C
+	scopedLog := log.WithName("getIdxcPodExecClient").WithValues("crName", cr.GetName(), "namespace", cr.GetNamespace())
+
+	// get the admin password from the namespace scoped secret
+	adminPwd, err := getAdminPasswrdFromSecret(client, cr)
+	if err != nil {
+		scopedLog.Error(err, "unable to get admin password")
+		return nil
+	}
+
+	return &IdxcPodExecClient{
+		client:        client,
+		cr:            cr,
+		targetPodName: targetPodName,
+		adminPwd:      adminPwd,
+		command:       fmt.Sprintf(applyIdxcBundleCmdStr, adminPwd),
+	}
+}
+
+// getIdxcPlayBookContext returns the idxc playbook context
+func getIdxcPlayBookContext(client splcommon.ControllerClient, cr splcommon.MetaObject, afwPipeline *AppInstallPipeline, podName string) *IdxcPlayBookContext {
+	return &IdxcPlayBookContext{
+		client:        client,
+		cr:            cr,
+		afwPipeline:   afwPipeline,
+		targetPodName: podName,
+		podExecClient: getIdxcPodExecClient(client, cr, applyIdxcBundleCmdStr, podName),
+	}
+}
+
+// getSHCPlayBookContext returns the shc playbook context
+func getSHCPlayBookContext(client splcommon.ControllerClient, cr splcommon.MetaObject, afwPipeline *AppInstallPipeline, podName string) *SHCPlayBookContext {
+	return &SHCPlayBookContext{
+		client:        client,
+		cr:            cr,
+		afwPipeline:   afwPipeline,
+		targetPodName: podName,
+	}
+}
+
+// getPlayBookContext returns the context for running playbook
+func getPlayBookContext(client splcommon.ControllerClient, cr splcommon.MetaObject, afwPipeline *AppInstallPipeline, podName string, kind string) PlayBookImpl {
+
+	switch kind {
+	case "ClusterMaster":
+		return getIdxcPlayBookContext(client, cr, afwPipeline, podName)
+	// TODO: gaurav - implement SHC playbook
+	//case "SearchHeadCluster":
+	//	return getSHCPlayBookContext(client, cr, afwPipeline, podName)
+	default:
+		return nil
+	}
+}
+
+// runPodExecCommand runs the commands related to idxc bundle push
+func (podExecClient *IdxcPodExecClient) runPodExecCommand() (string, string, error) {
+	if podExecClient == nil {
+		err := fmt.Errorf("podExecClient is nil")
+		return "", "", err
+	}
+
+	streamOptions := &remotecommand.StreamOptions{
+		Stdin: strings.NewReader(podExecClient.command),
+	}
+
+	return splutil.PodExecCommand(podExecClient.client, podExecClient.targetPodName, podExecClient.cr.GetNamespace(), []string{"/bin/sh"}, streamOptions, false, false)
+}
+
+// isBundlePushComplete checks the status of bundle push
+func (idxcPlayBookContext *IdxcPlayBookContext) isBundlePushComplete() bool {
+	scopedLog := log.WithName("isBundlePushComplete").WithValues("crName", idxcPlayBookContext.cr.GetName(), "namespace", idxcPlayBookContext.cr.GetNamespace())
+
+	stdOut, stdErr, err := idxcPlayBookContext.podExecClient.runPodExecCommand()
+	if err != nil || stdErr != "" {
+		scopedLog.Error(err, "show cluster-bundle-status failed", "stdout", stdOut, "stderr", stdErr)
+		return false
+	}
+
+	if !strings.Contains(stdOut, "cluster_status=None") {
```

Comment:
nit: instead of contains, is it possible to do a pattern match for `"cluster_status=None"` ?